### PR TITLE
Bump to go 1.18 (and restructure project layout)

### PR DESCRIPTION
**Pull request by sgreene570**
See title. Bump project build and go.mod to use go 1.18.

Also fix the project structure so it's flat and easier to build without context.

### CODE REVIEW

**Comment by sgreene570** [comment]
plz tag hacktoberfest

**Comment by sgreene570** [comment]
Fat fingered that one

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 default:
-  image: img.doom.fm/build/golang:latest-1.15.3-buster
+  image: img.doom.fm/build/golang:latest-1.18.7-buster
```
Comment (agargiulo) [inline .gitlab-ci.yml]:
```suggestion
  image: reg.agarg.me/build/golang:latest-1.18.7-bullseye
```

Comment (agargiulo) [reply .gitlab-ci.yml]: The old registry may or may not exist, but this is the current one I’d like to use.

**Review by agargiulo — CHANGES_REQUESTED** [review]
Thanks for doing this, minor nit on the docker registry but otherwise looks fine 😸

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ compile:
   extends:
     - .global-vars
   script:
-    - go build -ldflags "-extldflags '-static'" -o "${CI_PROJECT_DIR}/${RELEASE_BIN}" ./cmd/alsaRemoteControl
+    - go build -ldflags "-extldflags '-static'" -o "${CI_PROJECT_DIR}/${RELEASE_BIN}" .
```
Comment (agargiulo) [inline .gitlab-ci.yml:33]: Actually, just noticed:
if you don't rename `cmd/alsaRemoteControl/main.go -> main.go`
then you can build with, for example:
`GOOS=linux GOARCH=arm GOARM=6 go build ./cmd/alsaRemoteControl`
which gives you a default binary file of `alsaRemoteControl`

So I kind of like that approach

Comment (sgreene570) [reply .gitlab-ci.yml:33]: The binary name is set by `go.mod` actually. So running `go build .` in the project root prior to this PR results in a no-op believe it or not. Thats what I was trying to fix.

Comment (agargiulo) [reply .gitlab-ci.yml:33]: Oh, hmm, I guess the name is the same for both, but I have seen the `cmd/<binName>` approach in several other projects and it seems nice.

Comment (sgreene570) [reply .gitlab-ci.yml:33]: With this changes in place, running `go build .` in the project root results in the `alsaRemoteControl` binary being outputted to the project root by default.

Comment (agargiulo) [reply .gitlab-ci.yml:33]: I think you need that for `go install`? but that might be dated info, I can't recall

Comment (sgreene570) [reply .gitlab-ci.yml:33]: Yea, theres the go install factor, but I think its common practice to always have some main entry point in the root of your proj dir if your proj is directly executable?

Comment (sgreene570) [reply .gitlab-ci.yml:33]: We could go back and fourth but TBH ive seen a main.go in the proj root more often then not. if you dont like it then we can revise the PR :)

Comment (agargiulo) [reply .gitlab-ci.yml:33]: Gotcha, that's a fair point. This is fine as you have it then.

**Review by agargiulo — APPROVED** [review]

**Comment by sgreene570** [comment]
@agargiulo can you tag this `hacktoberfest-accepted`? IDK if the `hacktoberfest` label is enough

**Comment by sgreene570** [comment]
or actually merging it is just as good, nvmd.

**Comment by sgreene570** [comment]
hacktoberfest requirements nowadays are whack

**Pull request merged**